.github/workflows/release.yml

```diff
@@ -9,69 +8,115 @@ on:
             - release/*
 
 jobs:
-
-    build:
-
-        runs-on: ubuntu-latest
-
-        steps:
-
-            - uses: actions/checkout@v3
-
-            - name: Set up Python 3.10
-              uses: actions/setup-python@v3
-              with:
-                  python-version: '3.10'
-
-            - name: Install pypa/build
-              run: python -m pip install build
-
-            - name: Build a binary wheel and a source tarball
-              run: >-
-                  python -m
-                  build
-                  --sdist
-                  --wheel
-                  --outdir dist/
-
-            - name: Upload distribution artifact
-              uses: actions/upload-artifact@v3
-              with:
-                  name: release
-                  path: dist/
-
-    publish:
-
-        needs: [build]
-        runs-on: ubuntu-latest
-
-        steps:
-            - uses: actions/download-artifact@v3
-              name: Download distribution artifact
-              with:
-                  name: release
-                  path: dist/
-
-            - name: Publish distribution on Test PyPI
-              uses: pypa/gh-action-pypi-publish@release/v1
-              if: startsWith(github.ref, 'refs/heads/release/')
-              with:
-                  user: __token__
-                  password: ${{ secrets.TEST_PYPI_API_TOKEN }}
-                  repository_url: https://test.pypi.org/legacy/
-                  skip_existing: true
-
-            - uses: softprops/action-gh-release@v0.1.14
-              name: Create release
-              if: startsWith(github.ref, 'refs/tags/v')
-              with:
-                  files: |
-                      dist/*
-                  generate_release_notes: true
-
-            - name: Publish distribution on PyPI
-              uses: pypa/gh-action-pypi-publish@release/v1
-              if: startsWith(github.ref, 'refs/tags/v')
-              with:
-                  user: __token__
-                  password: ${{ secrets.PYPI_API_TOKEN }}
+  build:
+    name: Build distribution 📦
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.x"
+    - name: Install pypa/build
+      run: >-
+        python3 -m
+        pip install
+        build
+        --user
+    - name: Build a binary wheel and a source tarball
+      run: python3 -m build
+    - name: Store the distribution packages
+      uses: actions/upload-artifact@v3
+      with:
+        name: python-package-distributions
+        path: dist/
+
+  publish-to-pypi:
```

Comment:
Why do we need to change the overall strategy for publishing to pypi here? The previous publishing pipeline was running well.

Reply:
If I remember correctly, it failed, partially because the pypi token by pablo got expired. This version should be ok though

Reply:
Good question. We can revert back if it doesn't work